temporalio/contrib/openai_agents/_temporal_openai_agents.py

```diff
@@ -270,7 +276,8 @@ def configure_worker(self, config: WorkerConfig) -> WorkerConfig:
         ]
         return super().configure_worker(config)
 
-    async def run_worker(self, worker: Worker) -> None:
+    @asynccontextmanager
+    async def run_worker(self) -> AsyncIterator[None]:
```

Comment:
People need to be able to access the worker and its options IMO. Running a replayer is not like running a worker and should not be combined with it IMO. What if it only makes sense for me to do something for activities or Nexus operations? If we want the default implementation of `run_worker` and `run_replayer` to delegate to another method called `run_worker_or_replayer` or something like that.